contribs/gmf/src/directives/mobilegeolocation.js

```diff
@@ -0,0 +1,206 @@
+     * @param {angular.Attributes} attrs Attributes.
+     */
+    function(element, attrs) {
+      var templateUrl = attrs['gmfMobilegeolocationTemplateUrl'];
+      return goog.isDef(templateUrl) ?
+          templateUrl :
+          gmf.mobilegeolocationTemplateUrl;
+    });
+
+
+/**
+ * Provide a "mobile geolocation" directive.
+ *
+ * @example
+ * <gmf-mobile-geolocation
+ *     gmf-mobile-geolocation-map="ctrl.map"
+ *     gmf-mobile-geolocation-options="ctrl.mobileGeolocationOptions">
+ * </gmf-mobile-geolocation>
+ *
+ * @param {string|function(!angular.JQLite=, !angular.Attributes=)}
+ *     gmfMobilegeolocationTemplateUrl Template URL for the directive.
+ * @return {angular.Directive} The Directive Definition Object.
+ * @ngInject
+ * @ngdoc directive
+ * @ngname gmfMobileGeolocation
+ */
+gmf.mobileGeolocationDirective = function(gmfMobilegeolocationTemplateUrl) {
+  return {
+    restrict: 'E',
+    scope: {
+      'getMobileMapFn': '&gmfMobileGeolocationMap',
+      'getMobileGeolocationOptionsFn': '&gmfMobileGeolocationOptions'
+    },
+    controller: 'GmfMobileGeolocationController',
+    controllerAs: 'ctrl',
+    templateUrl: gmfMobilegeolocationTemplateUrl
+  };
+};
+
+
+gmfModule.directive('gmfMobileGeolocation', gmf.mobileGeolocationDirective);
+
+
+
+/**
+ * @constructor
+ * @param {angular.Scope} $scope The directive's scope.
+ * @param {ngeo.DecorateGeolocation} ngeoDecorateGeolocation Decorate
+ *     Geolocation service.
+ * @param {ngeo.FeatureOverlayMgr} ngeoFeatureOverlayMgr The ngeo feature
+ *     overlay manager service.
+ * @export
+ * @ngInject
+ * @ngdoc controller
+ * @ngname GmfMobileGeolocationController
+ */
+gmf.MobileGeolocationController = function($scope, ngeoDecorateGeolocation,
+    ngeoFeatureOverlayMgr) {
+
+  var map = $scope['getMobileMapFn']();
+  goog.asserts.assertInstanceof(map, ol.Map);
+
+  /**
+   * @type {!ol.Map}
+   * @private
+   */
+  this.map_ = map;
+
+  var options = $scope['getMobileGeolocationOptionsFn']() || {};
+  goog.asserts.assertObject(options);
+
+  /**
+   * @type {ngeo.FeatureOverlay}
+   * @private
+   */
+  this.featureOverlay_ = ngeoFeatureOverlayMgr.getFeatureOverlay();
+
+  /**
+   * @type {ol.Geolocation}
+   * @private
+   */
+  this.geolocation_ = new ol.Geolocation({
+    projection: map.getView().getProjection()
+  });
+
+  /**
+   * @type {ol.Feature}
+   * @private
+   */
+  this.positionFeature_ = new ol.Feature(new ol.geom.Point([0, 0]));
+
+  if (options.positionFeatureStyle) {
+    this.positionFeature_.setStyle(options.positionFeatureStyle);
+  }
+
+  /**
+   * @type {ol.Feature}
+   * @private
+   */
+  this.accuracyFeature_ = new ol.Feature();
+
+  if (options.accuracyFeatureStyle) {
+    this.accuracyFeature_.setStyle(options.accuracyFeatureStyle);
+  }
+
+  /**
+   * @type {number|undefined}
+   * @private
+   */
+  this.zoom_ = options.zoom;
+
+  this.geolocation_.on('change:accuracyGeometry', function() {
```

Comment:
Use `goog.events.listen` to register on a listener instead of `on`.
Also, use the ol3 constant to refer to an event instead of a string.

Example: https://github.com/Geoportail-Luxembourg/geoportailv3/blob/master/geoportailv3/static/js/measure/measuredirective.js#L212